sycl/include/CL/sycl/marray.hpp

```diff
@@ -0,0 +1,349 @@
+                                 marray<bool, NumElements>>::type              \
+  operator RELLOGOP(const marray &Lhs, const marray &Rhs) {                    \
+    marray<bool, NumElements> Ret;                                             \
+    for (size_t I = 0; I < NumElements; ++I) {                                 \
+      Ret[I] = Lhs[I] RELLOGOP Rhs[I];                                         \
+    }                                                                          \
+    return Ret;                                                                \
+  }                                                                            \
+  template <typename T, typename BaseT = DataT>                                \
+  friend typename std::enable_if<std::is_convertible<T, DataT>::value &&       \
+                                     std::is_integral<T>::value &&             \
+                                     std::is_integral<BaseT>::value,           \
+                                 marray<bool, NumElements>>::type              \
+  operator RELLOGOP(const marray &Lhs, const T &Rhs) {                         \
+    return Lhs RELLOGOP marray(static_cast<const DataT &>(Rhs));               \
+  }
+
+  __SYCL_RELLOGOP(==)
+  __SYCL_RELLOGOP(!=)
+  __SYCL_RELLOGOP(>)
+  __SYCL_RELLOGOP(<)
+  __SYCL_RELLOGOP(>=)
+  __SYCL_RELLOGOP(<=)
+
+  __SYCL_RELLOGOP_INTEGRAL(&&)
+  __SYCL_RELLOGOP_INTEGRAL(||)
+
+#undef __SYCL_RELLOGOP
+#undef __SYCL_RELLOGOP_INTEGRAL
+
+#ifdef __SYCL_UOP
+#error "Undefine __SYCL_UOP macro"
+#endif
+
+#define __SYCL_UOP(UOP, OPASSIGN)                                              \
+  friend marray &operator UOP(marray &Lhs) {                                   \
+    Lhs OPASSIGN 1;                                                            \
+    return Lhs;                                                                \
+  }                                                                            \
+  friend marray operator UOP(marray &Lhs, int) {                               \
+    marray Ret(Lhs);                                                           \
+    Lhs OPASSIGN 1;                                                            \
+    return Ret;                                                                \
+  }
+
+  __SYCL_UOP(++, +=)
+  __SYCL_UOP(--, -=)
+#undef __SYCL_UOP
+
+  // Available only when: dataT != cl_float && dataT != cl_double
+  // && dataT != cl_half
+  template <typename T = DataT>
+  friend typename std::enable_if<std::is_integral<T>::value, marray>::type
+  operator~(marray &Lhs) {
+    marray Ret;
+    for (size_t I = 0; I < NumElements; ++I) {
+      Ret[I] = ~Lhs[I];
+    }
+    return Ret;
+  }
+
+  friend marray<bool, NumElements> operator!(marray &Lhs) {
+    marray<bool, NumElements> Ret;
+    for (size_t I = 0; I < NumElements; ++I) {
+      Ret[I] = !Lhs[I];
+    }
+    return Ret;
+  }
+
+  friend marray operator+(marray &Lhs) {
```

Comment:
It looks like you are up to date to our latest bug fixes! Good. :-)
But there is a missing `const`